clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -32,8 +33,29 @@
     public long nanoseconds();
 
     /**
+     * Returns time in milliseconds as does System.currentTimeMillis(),
+     * but uses elapsed time from an arbitrary epoch more like System.nanoTime().
+     * The difference is that if somebody changes the system clock,
+     * Time.currentElapsedTime will change but nanoTime won't.
+     * @return  The time in milliseconds from some arbitrary point in time.
+     */
+    public long currentElapsedTime();
```

Comment:
I agree that this is useful. I think there are a number of places where we use `milliseconds()` where we should really be using a method like this one (I noticed that recently). It may be worth thinking about the naming so that the differences are clear.